LongLookupTables/get_long_lookup_tables.py

```diff
@@ -0,0 +1,136 @@
+# import tqdm
+import os
+import json
+
+from LongLookupTables.make_long_lookup_tables import make_long_lookup_tables
+from utils.get_default_params import get_default_params
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+
+# Dictionary that maps input name of lookup table dataset
+# to directory where it is located
+name2dir = {
+    # Lookup tables with training up to 3 compositions
+    "long_lookup": "LongLookupTables",
+
+    # Lookup tables with training 1 2 and 4 compositions (i.e jumping 3)
+    # - currently does not have appropriate generation.txt
+    # "long lookup jump": "LongLookupTablesJump",
+
+    # long lookup tables with a iniital training file without t7 and t8
+    # and then adding uncomposed t7 and t8 with all the rest
+    "long_lookup_oneshot": "LongLookupTablesOneShot",
+
+    # reverse long lookup table (i.e right to left hard attention)
+    "long_lookup_reverse": "LongLookupTablesReverse",
+
+    # noisy long lookup table where between each "real" table there's one noisy
+    # one. THe hard attention is thus a diagonal wich is less steep
+    "long_lookup_intermediate_noise": "LongLookupTablesIntermediateNoise",
+
+    # noisy long lookup table with a special start token saying when are the
+    # "real tables" starting. THe hard attention is thus a diagonal that
+    # starts at some random position.
+    "noisy_long_lookup_single": "NoisyLongLookupTablesSingle",
+
+    # noisy long lookup table where there are multiple start token and only
+    # the last one really counts
+    "noisy_long_lookup_multi": "NoisyLongLookupTablesMulti"
+}
+
+
+def get_long_lookup_tables(name, is_small=False, is_mini=False, longer_repeat=5):
+    """
+    Return the wanted lookup dataset information, downloads or generates
+    it if it is not already present
+
+    Args:
+        name ({"long_lookup", "long_lookup_oneshot", "long_lookup_reverse", 
+            "noisy_long_lookup_multi", "noisy_long_lookup_single",
+            "long_lookup_intermediate_noise"}) name of the long lookup task to get.
+
+        is_small (bool, optional): whether to run a smaller verson of the task.
+            Used for getting less statistically significant results.
+        is_mini (bool, optional): whether to run a smaller verson of the task.
+            Used for testing purposes.
+        longer_repeat (int, optional): number of longer test sets. 
+            - note if data is already generated with a certain longer repeat 
+            - If a longer repeat is called then the return paths will not exist
+            - In this case either delete the data folder from the specific LongLookup set
+            - Or implement a check to extend it and regenerate with higher longer_repeat
+
+    Returns:
+        task arguments (to be passed to instantiate a Task Object)
+    """
+
+    print("Getting Lookup Table")
```

Comment:
DOn't use prints but logs

Reply:
Changed to logs and removed prints so that you can pass in an optional logger to the LongLookupTask object.

Reply:
@YannDubs Does this make sense for logging?
```
import logging 

class LongLookupTask (Task):
        def __init__(self, name, is_small=False, is_mini=False, longer_repeat=5, logger=None):
                    logger = logger or logging.getLogger(__name__)
       ...
```
And then using `logger.info("...")`.